scipy/stats/_stats_py.py

```diff
@@ -10675,6 +10675,148 @@ def first_order(t):
     return res.root
 
 
+def lmoments_iv(sample, order, axis, sorted, standardize):
+
+    sample = np.asarray(sample)
+    message = "`sample` must be an array of real numbers."
+    if np.issubdtype(sample.dtype, np.integer):
+        sample = sample.astype(np.float64)
+    if not np.issubdtype(sample.dtype, np.floating):
+        raise ValueError(message)
+
+    message = "`order` must be an array of positive integers."
+    order = np.asarray(order)
+    if (not np.issubdtype(order.dtype, np.integer) or np.any(order <= 0)):
+        raise ValueError(message)
+
+    axis = np.asarray(axis)[()]
+    message = "`axis` must be an integer."
+    if not np.issubdtype(axis.dtype, np.integer) or axis.ndim != 0:
+        raise ValueError(message)
+
+    sorted = np.asarray(sorted)[()]
+    message = "`sorted` must be True or False."
+    if not np.issubdtype(sorted.dtype, np.bool_) or sorted.ndim != 0:
+        raise ValueError(message)
+
+    standardize = np.asarray(standardize)[()]
+    message = "`standardize` must be True or False."
+    if not np.issubdtype(standardize.dtype, np.bool_) or standardize.ndim != 0:
+        raise ValueError(message)
+
+    sample = np.moveaxis(sample, axis, -1)
+    sample = np.sort(sample, axis=-1) if not sorted else sample
+
+    return sample, order, axis, sorted, standardize
+
+
+def _br(x, *, r=0):
+    n = x.shape[-1]
+    x = np.expand_dims(x, axis=-2)
+    x = np.broadcast_to(x, x.shape[:-2] + (len(r), n))
+    x = np.triu(x)
+    j = np.arange(n, dtype=x.dtype)
+    n = np.asarray(n, dtype=x.dtype)[()]
+    return (np.sum(special.binom(j, r[:, np.newaxis])*x, axis=-1)
+            / special.binom(n-1, r) / n)
+
+
+def _prk(r, k):
+    return (-1)**(r-k)*special.binom(r, k)*special.binom(r+k, k)
+
+
+def _lmoment_outputs(kwds):
+    # should share this with `moment`, but to do that we need
+    # to change argument `moment` to `order`
+    order = np.atleast_1d(kwds.get('order', [1, 2, 3, 4]))
+    message = "`order` must be a scalar or a non-empty 1D array."
+    if order.size == 0 or order.ndim > 1:
+        raise ValueError(message)
+    return len(order)
+
+
+@_axis_nan_policy_factory(  # noqa: E302
+    _moment_result_object, n_samples=1, result_to_tuple=lambda x: (x,),
+    n_outputs=_lmoment_outputs
+)
+def lmoment(sample, order=[1, 2, 3, 4], *, axis=0, sorted=False, standardize=True):
```

Comment:
should we include type annotations on new functions? I looked in the core dev docs and didn't see anything explicit but I noticed some new function signatures seem to have them, or perhaps this is writer's choice? (my preference is for type annotations if possible but I don't want to be going against decisions already made that I might've missed.

Reply:
There has not been an explicit decision. I don't add type annotations, but others are welcome to do so in a follow-up PR.